### PR TITLE
Revamp the sign in and register pages

### PR DESCRIPTION
**Pull request by janfaracik**
**Sign in - Before**
image

**After**
image

image

**Register - Before**
image

**After**
image

image

This PR introduces bolder, more visual sign in/register pages. They adopt the same starburst/gradient stylings from the About page and Design Library.

Also removes the inline JS in the register page, resolving https://issues.jenkins.io/browse/JENKINS-71039.

### Testing done

* User can register
* User can sign in
* Error messages display correctly

### Proposed changelog entries

- Revamp the sign in and register pages. Add support for browser-native themes like darkmode.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7872"></a>

### CODE REVIEW

**Comment by timja** [comment]
Worth posting this on community.Jenkins.io for feedback?

**Comment by janfaracik** [comment]
> Worth posting this on community.Jenkins.io for feedback?

I've posted it https://community.jenkins.io/t/an-updated-sign-in-and-register-page/6982

What are your thoughts/feedback?

File: war/src/main/scss/pages/sign-in-register.scss
Comment (NotMyFault) [inline war/src/main/scss/pages/sign-in-register.scss:90]: Stylelint expects a kebab case-like pattern.

**Review by NotMyFault — COMMENTED** [review]
I like the proposed design a lot, but I'm not a big fan of the new inline JS, such as https://github.com/jenkinsci/jenkins/pull/7872/files#diff-b1c3edf5abaac4d83594131e9e59e0ccdc259c4800b88cecd6e281f71ba4efb2R270, causing the need of follow-up PRs.
We generally avoid introducing new [CSP](https://www.jenkins.io/doc/developer/security/csp/) violations.

**Comment by janfaracik** [comment]
> I like the proposed design a lot, but I'm not a big fan of the new inline JS, such as https://github.com/jenkinsci/jenkins/pull/7872/files#diff-b1c3edf5abaac4d83594131e9e59e0ccdc259c4800b88cecd6e281f71ba4efb2R270, causing the need of follow-up PRs.
> 
> We generally avoid introducing new [CSP](https://www.jenkins.io/doc/developer/security/csp/) violations.

Agreed, want me to split the JS as part of this PR?

**Comment by NotMyFault** [comment]
> Agreed, want me to split the JS as part of this PR?

Preferably, if that is not too much work? At least the part you added 👀

**Comment by janfaracik** [comment]
> > Agreed, want me to split the JS as part of this PR?
> 
> Preferably, if that is not too much work? At least the part you added 👀

Happy to do so 🙌

**Comment by yaroslavafenkin** [comment]
This looks like it's also addressing https://issues.jenkins.io/browse/JENKINS-71039, is that correct?

**Comment by NotMyFault** [comment]
> This looks like it's also addressing https://issues.jenkins.io/browse/JENKINS-71039, is that correct?

Seems so, if this PR is merged, there's no inline JS left in `signup.jelly`.

**Review by yaroslavafenkin — APPROVED** [review]
Tested manually, functionally seems alright to me.
Looks fine security wise too, thanks a lot for improving CSP situation!

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
Causes [JENKINS-71238](https://issues.jenkins.io/browse/JENKINS-71238).

**Comment by lemeurherve** [comment]
Also [JENKINS-71253](https://issues.jenkins.io/browse/JENKINS-71253)

**Comment by daniel-beck** [comment]
Causes [JENKINS-71254](https://issues.jenkins.io/browse/JENKINS-71254)

**Pull request merged**
